tests/test_smoke.py

```diff
@@ -723,7 +723,7 @@ def test_spot_recovery():
             f'sky spot launch --cloud aws --region {region} -n {name} "echo SKYPILOT_JOB_ID: \$SKYPILOT_JOB_ID; sleep 1000"  -y -d',
             'sleep 300',
             f's=$(sky spot queue); printf "$s"; echo; echo; printf "$s" | grep {name} | head -n1 | grep "RUNNING"',
-            f'RUN_ID=$(sky spot logs -n {name} --no-follow | grep SKYPILOT_JOB_ID | cut -d: -f2); echo $RUN_ID',
+            f'RUN_ID=$(sky spot logs -n {name} --no-follow | grep SKYPILOT_JOB_ID | cut -d: -f2); echo $RUN_ID | tee /tmp/{name}-run-id',
```

Comment:
Why isn't the previous way sufficient? Is it because it's not an `export`?

If this is the case why did the previous L737 pass? Is it because it was grepping an empty string?

Reply:
I think the main reason for that is we are using `subprocess.run` for each line of the command, causing the `RUN_ID` env var is not shared across multiple commands as each command is running in its own shell (making `export` not sufficient either).

Yes, the previous test passed due to greping empty string.